src/types/git.ts

```diff
@@ -0,0 +1,31 @@
+export enum SecretType {
+  NO_AUTH,
+  BASIC_AUTH,
+  SSH,
+  PERSONAL_ACCESS_TOKEN,
+  OAUTH,
+}
+
+export interface GitSource {
+  url: string;
+  secretType?: SecretType;
+  secretContent?: any;
+  ref?: string;
+  contextDir?: string;
+  devfilePath?: string;
+  dockerfilePath?: string;
+}
+
+export enum GitProvider {
```

Comment:
Maybe we only need this and not the others?

Reply:
Yes, I thought to clean up unnecessary files/methods at the end of I finish all pages. Will it be okay?